crates/bevy_asset/src/lib.rs

```diff
@@ -1976,6 +1976,391 @@ mod tests {
         });
     }
 
+    #[test]
+    fn error_on_immediate_load_of_self_dependency() {
+        let (mut app, dir, _source_events) = create_app_with_source_event_sender();
+        let asset_server = app.world().resource::<AssetServer>().clone();
+
+        dir.insert_asset_text(Path::new("abc.cool.ron"), "");
```

Comment:
Can we use a different extension, just because in a lot of these asset tests `cool.ron` refers to `CoolText` assets. Whatever extension you want!

Repeat below.